### PR TITLE
Boost: Fix overlay overlapping issues and update hard-coded post types

### PR DESCRIPTION
**Pull request by dilirity**
Fixes https://github.com/Automattic/boost-cloud/issues/296 and https://github.com/Automattic/boost-cloud/issues/302

## Proposed changes:

* Fix overlays showing under the next section;
* Update list of post types for "Other" group in ISA UI, to be fetched from the current website.
image

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

no

## Testing instructions:

* Setup this version of Boost;
* Make sure overlays (Score explanation and "Other" group from ISA UI) are showing properly;
* Make sure that the custom post types listed in the "Other" group tooltip are the correct ones.

### CODE REVIEW

**Review by donnchawp — DISMISSED** [review]
It works!

File: projects/plugins/boost/app/admin/class-admin.php [hunk 1]
```diff
@@ -128,7 +128,12 @@ public function enqueue_scripts() {
 		wp_localize_script(
 			$admin_js_handle,
 			'Jetpack_Boost',
-			$this->config->constants()
+			array_merge(
+				$this->config->constants(),
+				array(
+					'siteCustomPostTypes' => $this->get_custom_post_types(),
+				)
```
Comment (thingalon) [inline projects/plugins/boost/app/admin/class-admin.php]: Is there a reason not to just add this in `Config::constants`?

**Review by thingalon — COMMENTED** [review]
Just a question --

File: projects/plugins/boost/app/admin/class-admin.php (hunk 1 above)
Comment (dilirity) [reply projects/plugins/boost/app/admin/class-admin.php]: Hm, not sure why I didn't add it there - makes the most sense. Done in [e89357e](https://github.com/Automattic/jetpack/pull/31811/commits/e89357e8c93ad9dc1956b4178c6afa095ff4828f).

**Review by donnchawp — APPROVED** [review]
Works as advertised.

**Comment by haqadn** [comment]
Cherry-picked a1982b9884

**Pull request merged**